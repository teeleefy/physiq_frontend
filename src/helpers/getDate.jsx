function getDate() {
    const today = new Date();
    const year = today.getFullYear();
    let month = today.getMonth() + 1;
    if(month<10){
        month = `0${month}`
    }
    let date = today.getDate();
    if(date<10){
        date = `0${date}`
    }
    return `${year}-${month}-${date}`;
  }

export default getDate;
