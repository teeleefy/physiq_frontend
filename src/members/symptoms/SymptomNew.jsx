import { Form, FormGroup, Input, Button, Label, Alert } from "reactstrap";
import { useState, useContext } from "react";
import {MemberContext} from "../../auth/UserContext.js";
import PhysiqApi from "../../Api.js";
import { useNavigate } from "react-router-dom";
import '../styles/Symptom.css'

function SymptomNew({getDate}){
    let navigate = useNavigate();
    const { currentMember } = useContext(MemberContext);
    const [formMessages, setFormMessages] = useState([]);
    const [updateSuccess, setUpdateSuccess] =useState();
    const [formData, setFormData] = useState({name: "", startDate: "", endDate: "", notes: ""});
    const todaysDate = getDate();

    async function addSymptom() {
        try {
          let newSymptom= await PhysiqApi.addMemberSymptom(currentMember.id, formData);
          return { success: true };
        } catch (errors) {
          console.error("Add Symptom failed", errors);
          return { success: false, errors };
        }
      }

    async function handleSubmit(evt){
        evt.preventDefault();
        // console.log(formData);
        let result = await addSymptom();
        
        if(result.success){
        //    alert("Saved Changes!") 
           setFormMessages(['Symptom Added!'])
           setUpdateSuccess(true);
           navigate("..", { relative: "path"});
        }
        else{
            setFormMessages(result.errors);
            setUpdateSuccess(false);
        }
    };
  
    /** Update local state w/curr state of input elem */
  
    const handleChange = evt => {
      const { name, value }= evt.target;
      setFormData(fData => ({
        ...fData,
        [name]: value
      }));
    };

    return(
        <>
            <Form className="SymptomNew-Form m-4" >
            <h1 className="Symptom-h1">Add Symptom:</h1>
            <p className="text-secondary">Fields marked with <span className="text-danger">*</span> are required.</p>
            <FormGroup>
                <Label className="Symptom-label" for="name" >
                    <b>Symptom Name</b><span className="text-danger">*</span>
                </Label>
                <Input
                id="name"
                name="name"
                value={formData.name}
                placeholder="Enter symptom name..."
                type="text"
                onChange={handleChange}
                />
            </FormGroup>
            <FormGroup className="Symptom-date-input">
                <Label className="Symptom-label" for="startDate">
                    <b>Start Date</b>
                </Label>
                <Input
                id="startDate"
                name="startDate"
                value={formData.startDate}
                max={todaysDate}
                placeholder="Enter start date..."
                type="date"
                onChange={handleChange}
                />
                </FormGroup>
            <FormGroup className="Symptom-date-input">
                <Label className="Symptom-label" for="endDate">
                    <b>End Date</b>
                </Label>
                <Input
                id="endDate"
                name="endDate"
                value={formData.endDate}
                max={todaysDate}
                placeholder="Enter end date..."
                type="date"
                onChange={handleChange}
                />
                </FormGroup>
            <FormGroup>
                <Label className="Symptom-label" for="notes">
                    <b>Notes</b>
                </Label>
                <Input
                id="notes"
                name="notes"
                value={formData.notes}
                placeholder="Enter notes..."
                type="text"
                onChange={handleChange}
                />
            </FormGroup>
                {formMessages.length
                    ? formMessages.map(msg => <Alert color={updateSuccess ? "success": "danger"}>{msg}</Alert>)
                    : null
                }
                
                <Button className="btn-dark" onClick={handleSubmit}>Add Symptom</Button>
            </Form>
        </>
    )}
    
export default SymptomNew;