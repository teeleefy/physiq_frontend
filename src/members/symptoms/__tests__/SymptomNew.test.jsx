import {renderWithRouterAndMemberContext, screen} from '../../../../__tests__/test-tools';
import getDate from "../../../helpers/getDate.jsx"
import SymptomNew from '../SymptomNew';

describe('SymptomNew', () => {
    it('renders the SymptomNew component', () => {
      renderWithRouterAndMemberContext(<SymptomNew getDate={getDate}/>);
      screen.debug(); 
    });

    it("matches snapshot", function() {
      const { asFragment } = renderWithRouterAndMemberContext(<SymptomNew getDate={getDate}/>);
      expect(asFragment()).toMatchSnapshot();
    });

  });