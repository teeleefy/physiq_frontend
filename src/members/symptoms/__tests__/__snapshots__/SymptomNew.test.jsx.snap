// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SymptomNew > matches snapshot 1`] = `
<DocumentFragment>
  <form
    class="SymptomNew-Form m-4"
  >
    <h1
      class="Symptom-h1"
    >
      Add Symptom:
    </h1>
    <p
      class="text-secondary"
    >
      Fields marked with 
      <span
        class="text-danger"
      >
        *
      </span>
       are required.
    </p>
    <div
      class="mb-3"
    >
      <label
        class="Symptom-label form-label"
        for="name"
      >
        <b>
          Symptom Name
        </b>
        <span
          class="text-danger"
        >
          *
        </span>
      </label>
      <input
        class="form-control"
        id="name"
        maxlength="100"
        name="name"
        placeholder="Enter symptom name..."
        type="text"
        value=""
      />
    </div>
    <div
      class="Symptom-date-input mb-3"
    >
      <label
        class="Symptom-label form-label"
        for="startDate"
      >
        <b>
          Start Date
        </b>
      </label>
      <input
        class="form-control"
        id="startDate"
        max="2024-08-06"
        name="startDate"
        placeholder="Enter start date..."
        type="date"
        value=""
      />
    </div>
    <div
      class="Symptom-date-input mb-3"
    >
      <label
        class="Symptom-label form-label"
        for="endDate"
      >
        <b>
          End Date
        </b>
      </label>
      <input
        class="form-control"
        id="endDate"
        max="2024-08-06"
        name="endDate"
        placeholder="Enter end date..."
        type="date"
        value=""
      />
    </div>
    <div
      class="mb-3"
    >
      <label
        class="Symptom-label form-label"
        for="notes"
      >
        <b>
          Notes
        </b>
      </label>
      <textarea
        class="form-control"
        id="notes"
        maxlength="250"
        name="notes"
        placeholder="Enter notes..."
      />
    </div>
    <p
      class="text-secondary"
    >
      0/250 Characters
    </p>
    <button
      class="btn-dark btn btn-secondary"
      type="button"
    >
      Add Symptom
    </button>
  </form>
</DocumentFragment>
`;
